lisp/ess-inf.el

```diff
@@ -1830,10 +1796,22 @@ far as possible and return -1."
     (goto-char pos)
     n))
 
+(defun ess-step (&optional skip)
+  "Move point to the end of SKIP (if given) and then step.
+Step forward one line or to the next code line depending on
+`ess-eval-empty'. SKIP can be 'function, 'paragraph, or 'line."
+  (cond
```

Comment:
And if it is not given?

Reply:
Then don't skip, just step. I think that's clear from the code but if you want me to add it to the docstring I can.

Reply:
I modified the docstring slightly to make this clearer.

Reply:
Splitting hairs and introducing semantic difference between skip and step for this small context is not worthy IMO. `maybe` is mostly used when the author is lazy enough not to think about better and more concrete name.  The argument `skip` is commonly used as a verb or a boolean variable and is a bit confusing as a noun here. 

I think `(defun ess-skip-thing (&optional thing)` where `thing` is either 'function 'paragraph or 'line captures the idea very clearly. People are used to `thing-at-point` function. The doc could clarify that after the skipping we go to the next code line.

Reply:
Since it really does two things I broke it into two functions `ess-skip-thing` and `ess-step-line`. Since they aren't inferior-specific I also moved them to `ess-utils` along with `ess-next-line` which `ess-step-line` calls.